.github/actions/lint/action.yml

```diff
@@ -0,0 +1,23 @@
+name: golangci-lint-k6
+description: Run golangci-lint with the k6 ruleset
+
+runs:
+  using: composite
+  steps:
+    - name: Install Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: 1.20.x
+        check-latest: true
+    - name: Retrieve golangci-lint version
+      shell: bash
+      run: |
+        echo "PWD=$(pwd)" >> $GITHUB_OUTPUT
+        echo "GolangCIVersion=$(head -n 1 "./.golangci.yml" | tr -d '# ')" >> $GITHUB_OUTPUT
+      id: getenv
+    - name: golangci-lint
+      uses: golangci/golangci-lint-action@v3
+      with:
+        version: ${{ steps.getenv.outputs.GolangCIVersion }}
+        only-new-issues: true
```

Comment:
Yes, I've also ran into weirdness with this action's `only-new-issues` not working exactly as `--new-from-rev` does on the CLI. :face_exhaling:

Could we make this value configurable then, so that each repo can decide whether they want to enable it or not?

That said, I think the fact that the action actually makes GH comments on the lines with linter issues is a major UX win. Nobody wants to waste time going to the job output to see what failed :sweat_smile: That was a major reason we decided to switch to the action, after all, since we used the binary for a long time before that.

So I think the shortcomings of `only-new-issues` not working always should be a motivator to fix the lint issues, not to abandon the action. :)

Reply:
> Could we make this value configurable then, so that each repo can decide whether they want to enable it or not?

Probably no point, since even if we make it configurable, it simply won't work in some cases... :disappointed: 

At the moment, it will work correctly with a `pull_request` dispatch. And it will work... if not correctly, than at least how you'd expect it, when we run this action on `master` / `main` branch - it will show all linter errors. 

However, if some project wants to reuse the action in another branch without a PR, and only show the new errors that are not present in `master`, it won't work now or even if we make `only-new-issues` configurable. Thankfully, we don't have cases like this yet, and I don't see a reason why we'd want that. But if we ever do, maybe we can make the `args:` value configurable and keep the `golangci/golangci-lint-action` action :thinking: 

Anyway, having played with this a bit now, I am fine to keep the current action, and I agree that the inline comments it provides are nice.

Reply:
>We might want to get rid of this action and use the golangci-lint binary directly, for more flexibility and fewer bugs.

If it is possible, I'd prefer to keep it as action since it also has a nice feature of commenting on the pull request. So UX is better there. 

>It also doesn't always work correctly, e.g. on push events in branches or when running with https://github.com/nektos/act, even with a pull_request event, probably because of https://github.com/golangci/golangci-lint-action/issues/531.

But do we have a concrete example of misbehavior on our codebase? :thinking:

Reply:
As I said, it doesn't work as we'd expect when testing with `act` or when running in a naked branch. But it works as we want it when running it in a pull request or `master` / `main`, which is what we actually use, and the UX is indeed better as you both point out. So I'll resolve this thread and keep the action.